### ISSUE TITLE
Chunked upload

### ISSUE THREAD

**Issue by jkufner** [PhotoBackup/client-android#46]
There are two reasons for uploading a file in chunks:
1. File upload may get interrupted so it is nice to continue when it ended
2. File upload size is often limited to few megabytes so this would avoid this limitation and would work for any file size (videos).

Idea is to send file in chunks of some reasonable size, one chunk per http request. Once all chunks are transfered, the completed file is renamed to original name.

Basically the one thing to add is offset and length of the chunk and seek before writing. A hash of the whole file would be nice to check to be sure no chunk is missing.

**Comment by stephanepechard** [comment]
Yeah, it's something I should have work on quite a long time ago...
Thanks to show interest in this, it will put it back on my todo list 👍
